File: frontend/src/api/video.ts
/**
 * Video session API client for Tavus avatar integration.
 */

import { apiClient } from "./client";

// Types matching backend/src/models/video.py

export type VideoSessionStatus = "created" | "active" | "ended" | "error";
export type SessionType = "chat" | "briefing" | "debrief" | "consultation";

export interface VideoSessionCreate {
  session_type?: SessionType;
  context?: string;
  custom_greeting?: string;
  lead_id?: string;
  audio_only?: boolean;
}

export interface VideoTranscriptEntry {
  id: string;
  video_session_id: string;
  speaker: string;
  content: string;
  timestamp_ms: number;
  created_at: string;
}

/** @deprecated Use VideoTranscriptEntry */
export type TranscriptEntry = VideoTranscriptEntry;

export interface VideoSession {
  id: string;
  user_id: string;
  tavus_conversation_id: string;
  room_url: string | null;
  status: VideoSessionStatus;
  session_type: SessionType;
  started_at: string | null;
  ended_at: string | null;
  duration_seconds: number | null;
  created_at: string;
  lead_id?: string;
  is_audio_only?: boolean;
  perception_analysis?: Record<string, unknown>;
  transcripts?: VideoTranscriptEntry[];
}

export interface VideoSessionListResponse {
  items: VideoSession[];
  total: number;
  limit: number;
  offset: number;
}

/**
 * Create a new video session with Tavus avatar.
 */
export async function createVideoSession(
  data: VideoSessionCreate
): Promise<VideoSession> {
  const response = await apiClient.post<VideoSession>("/video/sessions", data);
  return response.data;
}

/**
 * List video sessions for the current user.
 */
export async function listVideoSessions(params?: {
  limit?: number;
  offset?: number;
  session_type?: SessionType;
  status?: VideoSessionStatus;
}): Promise<VideoSessionListResponse> {
  const response = await apiClient.get<VideoSessionListResponse>(
    "/video/sessions",
    { params }
  );
  return response.data;
}

/**
 * Get a specific video session by ID.
 */
export async function getVideoSession(sessionId: string): Promise<VideoSession> {
  const response = await apiClient.get<VideoSession>(
    `/video/sessions/${sessionId}`
  );
  return response.data;
}

/**
 * End an active video session.
 */
export async function endVideoSession(sessionId: string): Promise<VideoSession> {
  const response = await apiClient.post<VideoSession>(
    `/video/sessions/${sessionId}/end`
  );
  return response.data;
}

File: backend/src/models/video.py
"""Video-related Pydantic models for ARIA.

This module contains all models related to video sessions, including
Tavus integration and transcript entries.
"""

from datetime import datetime
from enum import Enum
from typing import Any

from pydantic import BaseModel


class VideoSessionStatus(str, Enum):
    """Status of a video session."""

    CREATED = "created"
    ACTIVE = "active"
    ENDED = "ended"
    ERROR = "error"


class SessionType(str, Enum):
    """Type of video session."""

    CHAT = "chat"
    BRIEFING = "briefing"
    DEBRIEF = "debrief"
    CONSULTATION = "consultation"


class VideoSessionCreate(BaseModel):
    """Request model for creating a new video session."""

    session_type: SessionType = SessionType.CHAT
    context: str | None = None
    custom_greeting: str | None = None
    lead_id: str | None = None
    conversation_id: str | None = None
    audio_only: bool = False


class VideoSessionResponse(BaseModel):
    """Response model for video session data."""

    id: str
    user_id: str
    tavus_conversation_id: str
    room_url: str | None
    status: VideoSessionStatus
    session_type: SessionType
    started_at: datetime | None
    ended_at: datetime | None
    duration_seconds: int | None
    created_at: datetime
    lead_id: str | None = None
    conversation_id: str | None = None
    is_audio_only: bool = False
    perception_analysis: dict[str, Any] | None = None
    transcripts: list["TranscriptEntryResponse"] | None = None


class TranscriptEntryResponse(BaseModel):
    """Response model for transcript entry data."""

    id: str
    video_session_id: str
    speaker: str
    content: str
    timestamp_ms: int
    created_at: datetime


class VideoSessionListResponse(BaseModel):
    """Response model for paginated list of video sessions."""

    items: list[VideoSessionResponse]
    total: int
    limit: int
    offset: int


class VideoToolCallRequest(BaseModel):
    """Request model for executing a video tool call from Tavus CVI."""

    tool_name: str
    arguments: dict[str, Any] = {}
    conversation_id: str | None = None


class VideoToolCallResponse(BaseModel):
    """Response model for video tool call results."""

    tool_name: str
    result: str
    success: bool


# Rebuild model to resolve forward reference
VideoSessionResponse.model_rebuild()
